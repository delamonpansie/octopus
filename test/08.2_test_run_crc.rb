#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
rows_per_wal = 10
EOD
  end
end

Env.env_eval do |env|
  start
  c = connect
  13.times do
    c.insert [1]
  end

  puts `OCTOPUS_CAT_RUN_CRC=1 ./octopus --cat 00000000000000000010.xlog 2>/dev/null | sed 's/tm:[^ ]* //'` + "\n"
  FileUtils.rm "00000000000000000010.xlog"

  restart
  c = connect
  5.times do
    c.insert [1]
  end
  puts `OCTOPUS_CAT_RUN_CRC=1 ./octopus --cat 00000000000000000010.xlog 2>/dev/null | sed 's/tm:[^ ]* //'` + "\n"
end
