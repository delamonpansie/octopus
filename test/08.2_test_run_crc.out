# box.insert([1])
1

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

lsn:10 shard:0 run_crc:0x01904545 scn:10 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:11 shard:0 run_crc:0x24c23baf scn:11 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:12 shard:0 run_crc:0x2066566c scn:12 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:13 shard:0 run_crc:0xe329b616 scn:13 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:14 shard:0 run_crc:0x5f7b1390 scn:14 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

lsn:10 shard:0 run_crc:0x01904545 scn:10 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:11 shard:0 run_crc:0x24c23baf scn:11 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:12 shard:0 run_crc:0x2066566c scn:12 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:13 shard:0 run_crc:0xe329b616 scn:13 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">
lsn:14 shard:0 run_crc:0x5f7b1390 scn:14 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00">

